library/firewall_lib_facts.py

```diff
@@ -0,0 +1,204 @@
+    setting_options = [
+        _file[:-4] for _file in os.listdir(directory) if _file.endswith(".xml")
+    ]
+    if not detailed:
+        # for _file in os.listdir(directory):
+        #     if _file.endswith(".xml"):
+        #         setting_options.append(_file[:-4])
+        return setting_options
+    else:
+        setting_name = os.path.basename(directory)
+        settings = []
+        for _item in setting_options:
+            if setting_name == "zones":
+                element = fw.config().getZoneByName(_item).getSettings()
+                element_settings = element.getSettingsDict()
+            elif setting_name == "services":
+                element = fw.config().getServiceByName(_item).getSettings()
+                element_settings = element.getSettingsDict()
+            elif setting_name == "icmptypes":
+                element_settings = {}
+                element = fw.config().getIcmpTypeByName(_item).getSettings()
+                element_settings["version"] = element.getVersion()
+                element_settings["short"] = element.getShort()
+                element_settings["description"] = element.getDescription()
+                element_settings["destination"] = element.getDestinations()
+            elif setting_name == "helpers":
+                element_settings = {}
+                element = fw.config().getHelperByName(_item).getSettings()
+                element_settings["version"] = element.getVersion()
+                element_settings["short"] = element.getShort()
+                element_settings["description"] = element.getDescription()
+                element_settings["family"] = element.getFamily()
+                element_settings["module"] = element.getModule()
+                element_settings["port"] = element.getPorts()
+            elif setting_name == "ipsets":
+                element_settings = {}
+                element = fw.config().getIPSetByName(_item).getSettings()
+                element_settings["version"] = element.getVersion()
+                element_settings["short"] = element.getShort()
+                element_settings["description"] = element.getDescription()
+                element_settings["options"] = element.getOptions()
+                element_settings["entries"] = element.getEntries()
+            elif setting_name == "policies":
+                element = fw.config().getPolicyByName(_item).getSettings()
+                element_settings = element.getSettingsDict()
+            settings.append(element_settings)
+        return settings
+
+
+def main():
+
+    module_args = dict(detailed=dict(type="bool", default=False, required=False))
+
+    results = dict(changed=False, firewall_config=dict())
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    if not HAS_FIREWALLD:
+        module.fail_json(msg="firewalld not installed")
+
+    if module.check_mode:
+        module.exit_json(**results)
+
+    fw = FirewallClient()
```

Comment:
If you don't use `fw` in this context, maybe declare it in `config_to_dict` or `fetch_settings_from_dir`, and only if you need it?  e.g.
```python
  fw = None
  if I_need_fw:
      fw = FirewallClient()
```